operators/multiclusterobservability/pkg/rendering/renderer_mcoa_test.go

```diff
@@ -169,12 +195,214 @@ func TestRenderAddonDeploymentConfig(t *testing.T) {
+							Enabled: true,
+						},
+					},
+				},
+				UserWorkloads: &mcov1beta2.UserWorkloadCapabilitiesSpec{
+					Logs: mcov1beta2.UserWorkloadLogsSpec{
+						Collection: mcov1beta2.UserWorkloadLogsCollectionSpec{
+							ClusterLogForwarder: mcov1beta2.ClusterLogForwarderSpec{
+								Enabled: true,
+							},
+						},
+					},
+				},
+			},
+			expectConfig: func(t *testing.T, cma *addonv1alpha1.ClusterManagementAddOn) {
+				assert.Len(t, cma.Spec.InstallStrategy.Placements[0].Configs, 1)
+			},
+		},
+		{
+			name: "add traces configs when user workloads is enabled",
+			capabilities: &mcov1beta2.CapabilitiesSpec{
+				UserWorkloads: &mcov1beta2.UserWorkloadCapabilitiesSpec{
+					Traces: mcov1beta2.UserWorkloadTracesSpec{
+						Collection: mcov1beta2.OpenTelemetryCollectionSpec{
+							Collector: mcov1beta2.OpenTelemetryCollectorSpec{
+								Enabled: true,
+							},
+							Instrumentation: mcov1beta2.InstrumentationSpec{
+								Enabled: true,
+							},
+						},
+					},
+				},
+			},
+			expectConfig: func(t *testing.T, cma *addonv1alpha1.ClusterManagementAddOn) {
+				assert.Len(t, cma.Spec.InstallStrategy.Placements[0].Configs, 2)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &MCORenderer{
+				cr: &mcov1beta2.MultiClusterObservability{
+					Spec: mcov1beta2.MultiClusterObservabilitySpec{
+						Capabilities: tt.capabilities,
+					},
+				},
+			}
+
+			// Add deployment config to reflect real input
+			baseCMA := &addonv1alpha1.ClusterManagementAddOn{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"k1": "v1",
+					},
+				},
+				Spec: addonv1alpha1.ClusterManagementAddOnSpec{
+					InstallStrategy: addonv1alpha1.InstallStrategy{
+						Placements: []addonv1alpha1.PlacementStrategy{
+							{
+								Configs: []addonv1alpha1.AddOnConfig{},
+							},
+						},
+					},
+				},
+			}
+			// to kustomize resource
+			baseCMAUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(baseCMA)
+			assert.NoError(t, err)
+			kres := &kustomizeres.Resource{}
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(baseCMAUnstructured, kres)
+			assert.NoError(t, err)
+
+			res, err := r.renderClusterManagementAddOn(kres, "ns", map[string]string{"k2": "v2"})
+			assert.NoError(t, err)
+			assert.NotNil(t, res)
+
+			// check labels
+			assert.Len(t, res.GetLabels(), 2)
+
+			// check supportedConfigs
+			cma := &addonapiv1alpha1.ClusterManagementAddOn{}
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(res.Object, cma)
+			assert.NoError(t, err)
+
+			tt.expectConfig(t, cma)
+
+			// Check duplicated supportedConfigs
+			dups := make(map[string]struct{})
+			for _, cfg := range cma.Spec.SupportedConfigs {
+				key := cfg.ConfigGroupResource.Group + "/" + cfg.ConfigGroupResource.Resource
+				if _, ok := dups[key]; ok {
+					t.Errorf("duplicated supportedConfigs %s", key)
+				}
+				dups[key] = struct{}{}
+			}
```

Comment:
Why are we checking for duplicates? This is actually something that is [supported by the addon-framework](https://github.com/open-cluster-management-io/addon-framework/issues/263)

Reply:
The addon framework will fail at runtime, I'm just checking it earlier at unit-test time.